GPU/GPUTracking/TPCClusterFinder/PeakFinder.cxx

```diff
@@ -68,37 +62,40 @@ GPUd() bool PeakFinder::isPeakScratchPad(
     return false;
   }
 
+  // Ensure q has the same float->int->float conversion error
+  // as values in chargeMap, so identical charges are actually identical
+  q = PackedCharge(q).unpack();
+
+  int idx = N * partId;
   bool peak = true;
-  for (ushort i = 0; i < N; i++) {
-    Charge q = buf[N * partId + i].unpack();
-    peak &= (digit->charge > q) || (CfConsts::InnerTestEq[i] && digit->charge == q);
-  }
+  peak = peak && buf[idx + 0].unpack() <= q;
+  peak = peak && buf[idx + 1].unpack() <= q;
+  peak = peak && buf[idx + 2].unpack() <= q;
+  peak = peak && buf[idx + 3].unpack() <= q;
+  peak = peak && buf[idx + 4].unpack() < q;
+  peak = peak && buf[idx + 5].unpack() < q;
+  peak = peak && buf[idx + 6].unpack() < q;
+  peak = peak && buf[idx + 7].unpack() < q;
 
   return peak;
 }
 
 GPUd() bool PeakFinder::isPeak(
-  const Digit* digit,
-  GPUglobalref() const PackedCharge* chargeMap)
+  Charge myCharge,
+  const ChargePos& pos,
+  const Array2D<PackedCharge>& chargeMap)
 {
-  if (digit->charge <= QMAX_CUTOFF) {
+  if (myCharge <= QMAX_CUTOFF) {
     return false;
   }
 
-  const Charge myCharge = digit->charge;
-  const Timestamp time = digit->time;
-  const Row row = digit->row;
-  const Pad pad = digit->pad;
-
-  const GlobalPad gpad = Array2D::tpcGlobalPadIdx(row, pad);
-
   bool peak = true;
 
-#define CMP_NEIGHBOR(dp, dt, cmpOp)                           \
-  do {                                                        \
-    PackedCharge p = CHARGE(chargeMap, gpad + dp, time + dt); \
-    const Charge otherCharge = p.unpack();                    \
-    peak &= (otherCharge cmpOp myCharge);                     \
+#define CMP_NEIGHBOR(dp, dt, cmpOp)                  \
```

Comment:
This change triggers for some reason an internal compiler error in clang 10 C++ for OpenCL.
I am filing a bug report to clang, but I'll wait with merging until I get a response.